frontend/src/app/main/ui/ds/notifications/toast.cljs

```diff
@@ -10,36 +10,26 @@
    [app.main.style :as stl])
   (:require
    [app.main.ui.ds.foundations.assets.icon :as i]
+   [app.main.ui.ds.notifications.shared.notification-pill :refer [notification-pill*]]
    [rumext.v2 :as mf]))
 
-(def ^:private icons-by-level
-  {"info" i/info
-   "warning" i/msg-neutral
-   "error" i/delete-text
-   "success" i/status-tick})
-
 (def ^:private schema:toast
   [:map
    [:class {:optional true} :string]
-   [:level {:optional true}
-    [:maybe [:enum "info" "warning" "error" "success"]]]
+   [:type  {:optional true} [:maybe [:enum :toast :context]]]
+   [:level {:optional true} [:maybe [:enum :info :warning :error :success]]]
    [:on-close {:optional true} fn?]])
 
 (mf/defc toast*
   {::mf/props :obj
    ::mf/schema schema:toast}
-  [{:keys [class level children on-close] :rest props}]
-  (let [class (dm/str (stl/css-case :toast true
-                                    :toast-info (= level "info")
-                                    :toast-warning (= level "warning")
-                                    :toast-error (= level "error")
-                                    :toast-success (= level "success")) " " class)
-        icon-id (or (get icons-by-level level) i/msg-neutral)
-        props (mf/spread-props props {:class class})]
-    [:> "aside" props
-     [:*
-      [:> i/icon* {:icon-id icon-id :class (stl/css :icon)}]
-      children
+  [{:keys [class level type children on-close] :rest props}]
+  (let [class (dm/str class " " (stl/css-case :toast true))
+        level (or level :info)
+        type (or type :context)
+        props (mf/spread-props props {:class class :role "alert"})]
+    [:> "div" props
+     [:> notification-pill* {:level (keyword level) :type type} children]
```

Comment:
The `(keyword level)` is executed always, independently if the level is a string or keyword.
I think you should check if coming level is string on let

```clojure
(let [...
      level (if (string? level)
              (keyword level)
              (d/nilv level :info))
```

Reply:
But according to the schema, it must always be a keyword.

`[:level {:optional true} [:maybe [:enum :info :warning :error :success]]]`

You mean to set this validation because of this component can be used in JavaScript?

Reply:
I've added this validation for strings, but storybook (for instance) it works well using keywords as well.

Reply:
I mean, if you conde is always suppose to receive keyword, then  you don't need the `(keyword level)` because you are converting keyword to keyword... But as far as I understand this is done for make it more easy use from storybook, where it can receive strings there, in this case we need that check. It all depends if storybook is passing correct data or it passing strings for this.